spec/schedulers/AsapScheduler-spec.ts

```diff
@@ -288,4 +288,25 @@ describe('Scheduler.asap', () => {
       done();
     });
   });
+
+  it('should handle actions scheduled during flush before current action is rescheduled', (done) => {
+    const sandbox = sinon.createSandbox();
+
+    const result: string[] = [];
+    let reschedule = true;
+    function work(this: SchedulerAction<unknown>) {
+      result.push('work');
+      if (reschedule) {
+        asapScheduler.schedule(() => result.push('task 1'));
+        asapScheduler.schedule(() => result.push('task 2'));
```

Comment:
Can we add an assertion right here to make sure that the above tasks didn't execute synchronously in this spot? It's not so much that I think they did, it's that I want to make sure they're "scheduled" and don't just happen to mimic the proper outcome by virtue of when we checked.

Please do the same for the other test? Thanks, @ajafff!

Otherwise, this looks great.

Reply:
done

Reply:
@benlesh could you please review it again? Can't upgrade above 7.4.0 due to this problem.

Reply:
OMG. This is a serious bug and produces all kind of issues on applications that use animationFrameScheduler.
I didn't notice about this PR (and created a similar one myself), almost 2 years later! [!7444](https://github.com/ReactiveX/rxjs/pull/7444)

Can we please review one of them and merge it?